packages/image_picker/image_picker_macos/lib/image_picker_macos.dart

```diff
@@ -24,6 +30,42 @@ class ImagePickerMacOS extends CameraDelegatingImagePickerPlatform {
     ImagePickerPlatform.instance = ImagePickerMacOS();
   }
 
+  /// Sets [ImagePickerMacOS] to use [PHPicker](https://developer.apple.com/documentation/photokit/phpickerviewcontroller)
+  /// which is **only supported on macOS 13.0+**.
+  ///
+  /// Will fallback to [file_selector_macos](https://pub.dev/packages/file_selector_macos)
+  /// if [useMacOSPHPicker] is `false` or the macOS version doesn't support
+  /// this feature.
+  ///
+  /// Currently defaults to `false`.
+  ///
+  /// **Note**: This implementation depends on the photos in the [Photos for macOS App](https://www.apple.com/in/macos/photos/),
+  /// if the user didn't open the app or import any photos to the app,
+  /// they will see: `No photos` or `No Photos or Videos` message even if they
+  /// have them as files on their desktop.
+  ///
+  /// Supports picking an image, multi-image, video, media, and multiple media.
+  bool useMacOSPHPicker = false;
+
+  // TODO(EchoEllet): avoid using @visibleForTesting per https://github.com/flutter/flutter/blob/master/docs/contributing/Style-guide-for-Flutter-repo.md#avoid-using-visiblefortesting
+
+  /// Return `true` if the current macOS version supports [useMacOSPHPicker].
+  ///
+  /// The [PHPicker](https://developer.apple.com/documentation/photokit/phpickerviewcontroller)
+  /// is **supported on macOS 13.0+**
+  @visibleForTesting
```

Comment:
The PR description suggests that developers would want to provide this as a user-level opt-in, in which case they would need this information to know whether to show the option, so this should be public.